src/modules/currency/wallet/currency-wallet-files.js

```diff
@@ -66,74 +143,112 @@ export function setCurrencyWalletFiat (
-          type: 'CURRENCY_WALLET_NAME_CHANGED',
-          payload: { name, walletId }
-        })
-      }),
-
-    folder
-      .file('Currency.json')
-      .getText()
-      .then(text => JSON.parse(text))
-      .then(file =>
-        dispatch({
-          type: 'CURRENCY_WALLET_FIAT_CHANGED',
-          payload: {
-            fiatCurrencyCode: file.fiat
-              ? file.fiat
-              : 'iso:' + currencyFromNumber(file.num).code,
-            walletId
-          }
-        })
-      )
-      .catch(e =>
-        dispatch({
-          type: 'CURRENCY_WALLET_FIAT_CHANGED',
-          payload: { fiatCurrencyCode: 'iso:USD', walletId }
-        })
-      ),
+  return folder
+    .file('Currency.json')
+    .getText()
+    .then(text => {
+      const file = JSON.parse(text)
+      return file.fiat ? file.fiat : 'iso:' + currencyFromNumber(file.num).code
+    })
+    .catch(e => 'iso:USD')
+    .then((fiatCurrencyCode: string) => {
+      dispatch({
+        type: 'CURRENCY_WALLET_FIAT_CHANGED',
+        payload: { fiatCurrencyCode, walletId }
+      })
+      return fiatCurrencyCode
+    })
+}
+
+/**
+ * Loads the wallet name file.
+ */
+function loadNameFile (input: CurrencyWalletInput, folder) {
+  const walletId = input.props.id
+  const { dispatch } = input.props
 
+  return folder
+    .file('WalletName.json')
+    .getText()
+    .then(text => JSON.parse(text).walletName)
+    .catch(e => null)
+    .then((name: string | null) =>
+      dispatch({
+        type: 'CURRENCY_WALLET_NAME_CHANGED',
+        payload: { name, walletId }
+      })
+    )
+}
+
+/**
+ * Loads transaction metadata files.
+ */
+function loadTxFiles (input: CurrencyWalletInput, folder) {
+  const walletId = input.props.id
+  const { dispatch } = input.props
+  const walletCurrency = input.props.selfState.currencyInfo.currencyCode
+  const walletFiat = input.props.selfState.fiat
+
+  // Actually load the files:
+  const allFiles = Promise.all([
     // Transaction metadata:
     mapFiles(folder.folder('transaction'), file =>
       file
         .getText()
         .then(text => JSON.parse(text))
         .catch(e => null)
-    ).then(files => {
-      const out = {}
-      const jsons = files.filter(json => json != null && json.txid != null)
-      for (const json of jsons) {
-        out[json.txid] = json
-      }
-      return dispatch({
-        type: 'CURRENCY_WALLET_FILES_LOADED',
-        payload: { files: out, walletId }
-      })
-    })
+    ),
+
+    // Legacy transaction metadata:
+    mapFiles(folder.folder('Transactions'), file =>
+      file
+        .getText()
+        .then(text => JSON.parse(text))
+        .catch(e => null)
+    )
   ])
+
+  // Save the results to redux:
+  return allFiles.then(allFiles => {
+    const [newFiles, oldFiles] = allFiles
+
+    const out = {}
+    for (const json of newFiles) {
+      if (json == null || !json.txid) continue
+      out[json.txid] = json
+    }
+    for (const json of oldFiles) {
```

Comment:
This seems like you will overwrite data from newFiles with data from oldFiles. Since this runs in Edge, shouldn't you prioritize newFiles over oldFiles?